STY: Enforce ruff rules

DimitriPapadopoulos opened the pull request:
## Summary

Disable some rules and postpone fixes.

Requires #3689.

DimitriPapadopoulos committed:
STY: Apply ruff/flake8-bugbear rule B007

B007 Loop control variable not used within loop body

Co-authored-by: Chris Markiewicz <effigies@gmail.com>

DimitriPapadopoulos committed:
STY: Apply ruff/flake8-bugbear rule B018

B018 Found useless expression. Either assign it to a variable or remove it.

DimitriPapadopoulos committed:
STY: Apply ruff/flake8-bugbear rule B015

B015 Pointless comparison at end of function scope.
     Did you mean to return the expression result?

DimitriPapadopoulos committed:
STY: Apply ruff/refurb rule FURB154

FURB154 Use of repeated consecutive `global`

DimitriPapadopoulos committed:
STY: Apply ruff/Pylint rule PLE0101

PLE0101 Explicit return in `__init__`

DimitriPapadopoulos committed:
STY: Apply ruff/flake8-pytest-style rule PT006

PT006 Wrong type passed to first argument of `@pytest.mark.parametrize`; expected `tuple`

DimitriPapadopoulos committed:
STY: Apply ruff/flake8-pytest-style rule PT007

PT007 Wrong values type in `@pytest.mark.parametrize` expected `list` of `tuple`

DimitriPapadopoulos committed:
STY: Apply ruff/flake8-pytest-style rule PT014

PT014 Duplicate of test case in `@pytest_mark.parametrize`

DimitriPapadopoulos committed:
STY: Apply ruff/flake8-quotes rule Q000

Q000 Single quotes found but double quotes preferred

DimitriPapadopoulos committed:
STY: Apply ruff/flake8-quotes rule Q003

Q003 Change outer quotes to avoid escaping inner quotes

DimitriPapadopoulos committed:
STY: Apply ruff/pyupgrade rule UP008

UP008 Use `super()` instead of `super(__class__, self)`

DimitriPapadopoulos committed:
STY: Apply ruff/Pylint rule PLE1205

PLE1205 Too many arguments for `logging` format string

DimitriPapadopoulos committed:
STY: Enforce ruff rules

Disable some rules and postpone fixes.

DimitriPapadopoulos committed:
STY: Apply ruff/pyupgrade rule UP032

UP032 Use f-string instead of `format` call

Co-authored-by: Chris Markiewicz <effigies@gmail.com>

DimitriPapadopoulos committed:
STY: Apply ruff rule RUF100

RUF100 Unused blanket `noqa` directive

DimitriPapadopoulos committed:
Apply ruff/pycodestyle rule E741

E741 Ambiguous variable name

effigies commented on nipype/algorithms/modelgen.py:
```diff
@@ -597,7 +597,7 @@ def _concatenate_info(self, infolist):
             # info.[conditions, tmod] remain the same
             if info.onsets:
-                for j, val in enumerate(info.onsets):
+                for j, _ in enumerate(info.onsets):
                     if self.inputs.input_units == "secs":
                         onsets = np.array(
                             info.onsets[j]
```
And also in L607.

```suggestion
                for j, val in enumerate(info.onsets):
                    if self.inputs.input_units == "secs":
                        onsets = np.array(
                            val
```

effigies commented on nipype/algorithms/modelgen.py:
```diff
@@ -621,7 +621,7 @@ def _concatenate_info(self, infolist):
 
                 if hasattr(info, "amplitudes") and info.amplitudes:
-                    for j, val in enumerate(info.amplitudes):
+                    for j, _ in enumerate(info.amplitudes):
                         infoout.amplitudes[j].extend(info.amplitudes[j])
```
```suggestion
                    for j, val in enumerate(info.amplitudes):
                        infoout.amplitudes[j].extend(val)
```

effigies commented on nipype/algorithms/modelgen.py:
```diff
@@ -633,7 +633,7 @@ def _concatenate_info(self, infolist):
                 # assumes same ordering of regressors across different
                 # runs and the same names for the regressors
-                for j, v in enumerate(info.regressors):
+                for j, _ in enumerate(info.regressors):
                     infoout.regressors[j].extend(info.regressors[j])
```
```suggestion
                for j, v in enumerate(info.regressors):
                    infoout.regressors[j].extend(v)
```

effigies commented on nipype/interfaces/ants/registration.py:
```diff
@@ -204,11 +204,7 @@ def _transformation_constructor(self):
-            self.inputs.regularization_gradient_field_sigma,
-            self.inputs.regularization_deformation_field_sigma,
-        )
+        return f"--regularization {self.inputs.regularization}[{self.inputs.regularization_gradient_field_sigma},{self.inputs.regularization_deformation_field_sigma}]"
```
This doesn't strike me as more readable. Could do something like:

```suggestion
        gfsigma = self.inputs.regularization_gradient_field_sigma
        dfsigma = self.inputs.regularization_deformation_field_sigma
        return f"--regularization {self.inputs.regularization}[{gfsigma},{dfsigma}]"
```

Alternately, we could use format strings sensibly using `input_spec` fields and then `trait_get()` to populate them:

```suggestion
        fmt = "--regularization {regularization}[{regularization_gradient_field_sigma},{regularization_deformation_field_sigma}]"
        return fmt.format(**self.inputs.trait_get())
```

I would do this throughout. I don't think f-strings are improving things here, even if they might be very slightly more efficient.

DimitriPapadopoulos replied:
I agree. I'm not very happy with these changes. I can't decide whether I should keep UP0032 altogether and discard these changes, or apply the changes that make sense and silence the rest with `noqa`. What would you do?

effigies replied:
We could do a per-file ignore, if it's only a couple files that get really ugly. Then we can revisit them separately or never. `%` and `str.format` aren't going away, so there's no time bomb if we end up just saying "ANTs is too special for UP032".

If it's broad, we could set the ignore to everything in "nipype/interfaces".

effigies commented on nipype/interfaces/cmtk/nx.py:
```diff
@@ -550,14 +550,14 @@ def _run_interface(self, runtime):
         dicts = list()
-        for idx, key in enumerate(dict_measures.keys()):
-            for idxd, keyd in enumerate(dict_measures[key].keys()):
+        for key, value in dict_measures:
```
```suggestion
        for key, value in dict_measures.items():
```

effigies commented on nipype/interfaces/freesurfer/utils.py:
```diff
@@ -355,8 +355,7 @@ def _format_arg(self, name, spec, value):
                         raise ValueError(
-                            "Cannot create {} file with extension "
-                            "{}".format(value, ext)
+                            f"Cannot create {value} file with extension " f"{ext}"
```
```suggestion
                            f"Cannot create {value} file with extension {ext}"
```

effigies commented on nipype/interfaces/freesurfer/utils.py:
```diff
@@ -595,8 +594,7 @@ def _format_arg(self, name, spec, value):
                         raise ValueError(
-                            "Cannot create {} file with extension "
-                            "{}".format(value, ext)
+                            f"Cannot create {value} file with extension " f"{ext}"
```
```suggestion
                            f"Cannot create {value} file with extension {ext}"
```

effigies commented on nipype/interfaces/fsl/model.py:
```diff
@@ -1388,8 +1386,7 @@ def _run_interface(self, runtime):
         ]
-        for i in range(self.inputs.num_copes):
-            mat_txt += ["1"]
+        mat_txt.extend("1" for _ in range(self.inputs.num_copes))
```
```suggestion
        mat_txt.extend(["1"] * self.inputs.num_copes)
```

effigies commented on nipype/interfaces/fsl/model.py:
```diff
@@ -1411,16 +1408,15 @@ def _run_interface(self, runtime):
         ]
-        for i in range(self.inputs.num_copes):
-            grp_txt += ["1"]
+        grp_txt.extend("1" for _ in range(self.inputs.num_copes))
```
```suggestion
        grp_txt.extend(["1"] * self.inputs.num_copes)
```

effigies commented on nipype/interfaces/fsl/preprocess.py:
```diff
@@ -426,7 +426,7 @@ def _list_outputs(self):
                 # for multi-image segmentation there is one corrected image
                 # per input
-                for val, f in enumerate(self.inputs.in_files):
+                for val, _ in enumerate(self.inputs.in_files):
                     # image numbering is 1-based
                     outputs["restored_image"].append(
                         self._gen_fname(
```
```python
outputs["restored_image"].extend(
    self._gen_fname(suffix=f"_restore_{i + 1}", **_gen_fname_opts)
    for i in range(len(self.inputs.in_files))
)
```

effigies reviewed (commented):
10 comments, to avoid collapsing.

effigies commented:
Just realized this is still in draft. Will hold off on further review for now.

effigies commented on nipype/interfaces/fsl/preprocess.py:
```diff
@@ -454,7 +451,7 @@ def _list_outputs(self):
                 # for multi-image segmentation there is one bias field image
                 # per input
-                for val, f in enumerate(self.inputs.in_files):
+                for val, _ in enumerate(self.inputs.in_files):
                     # image numbering is 1-based
                     outputs["bias_field"].append(
                         self._gen_fname(
```
Same as above?

```suggestion
                outputs["bias_field"].extend(
                    self._gen_fname(suffix=f"_bias_{i}", **_gen_fname_opts
                    for i in range(1, len(self.inputs.in_files) + 1)
                )
```

And looks like there's another block to clean up around L470 below.

effigies commented on nipype/interfaces/spm/preprocess.py:
```diff
@@ -1911,7 +1911,7 @@ def _list_outputs(self):
         if isdefined(self.inputs.tissues):
             n_classes = len(self.inputs.tissues)
-        for i in range(n_classes):
+        for _ in range(n_classes):
             outputs["native_class_images"].append([])
             outputs["dartel_input_images"].append([])
             outputs["normalized_class_images"].append([])
```
```suggestion
        outputs["native_class_images"].extend([] for _ in range(n_classes))
        outputs["dartel_input_images"].extend([] for _ in range(n_classes))
        outputs["normalized_class_images"].extend([] for _ in range(n_classes))
        outputs["modulated_class_images"].extend([] for _ in range(n_classes))
```

effigies commented on nipype/interfaces/spm/preprocess.py:
```diff
@@ -2159,7 +2159,7 @@ def _list_outputs(self):
         if isdefined(self.inputs.tissues):
             n_classes = len(self.inputs.tissues)
-        for i in range(n_classes):
+        for _ in range(n_classes):
             outputs["native_class_images"].append([])
             outputs["dartel_input_images"].append([])
             outputs["normalized_class_images"].append([])
```
```suggestion
        outputs["native_class_images"].extend([] for _ in range(n_classes))
        outputs["dartel_input_images"].extend([] for _ in range(n_classes))
        outputs["normalized_class_images"].extend([] for _ in range(n_classes))
        outputs["modulated_class_images"].extend([] for _ in range(n_classes))
```

effigies reviewed (commented):
Small suggestions.

DimitriPapadopoulos committed:
Update ruff to 0.8.0

DimitriPapadopoulos committed:
STY: Apply ruff/flake8-bugbear rule B010

B010 Do not call `setattr` with a constant attribute value.
     It is not any safer than normal property access.

DimitriPapadopoulos committed:
STY: Ignore ruff/pycodestyle rule E741 for now

DimitriPapadopoulos committed:
STY: Apply ruff/Pyflakes rule F522

F522 `.format` call has unused named argument

DimitriPapadopoulos committed:
Apply ruff/Pyflakes rule F821

F821 Undefined name

DimitriPapadopoulos committed:
Apply ruff rule RUF100

RUF100 Unused blanket `noqa` directive
RUF100 Unused `noqa` directive

DimitriPapadopoulos committed:
Apply ruff/pyupgrade rule UP015

UP015 Unnecessary open mode parameters

DimitriPapadopoulos committed:
Apply ruff/pyupgrade rule UP025

UP025 Remove unicode literals from strings

effigies commented on nipype/pipeline/plugins/legacymultiproc.py:
```diff
@@ -64,7 +64,7 @@ def run_node(node, updatehash, taskid):
     try:
         result["result"] = node.run(updatehash=updatehash)
-    except:  # noqa: E722, intendedly catch all here
+    except:
```
I think in general we would want to catch cases of E722 instead of blanket ignoring.
